### ISSUE TITLE
Add new languages to context database

### ISSUE THREAD

**Issue by arminsandorf** [sergeyshushlyapin/Sitecore.FakeDb#161]
Hi Sergey,

first of all, thank you for your great work developing FakeDB :) It helps tremendously at daily work!

Now I run into the following issue:

By using the LanguageManager within FakeDB, I got only the 'en' language from the context database. Diving into your code, I found the FakeDataProvider.GetLanguages() method which always returns a new language collection containing only 'en'. 

To my understanding, adding a new language needs to change this method directly. My first idea was to add languages using a settings node within the App.config:

```
<settings>
	<languages>
		<language name="en-GB" />
		<language name="de-DE" />
		<language name="de-CH" />
		<language name="fr-CH" />
		...
	</languages>
</settings>
```

... and reading this node from the method:

```
public override LanguageCollection GetLanguages(CallContext context)
{
	var settings = // read settings here ...
	var result = new LanguageCollection();
	foreach (var lang in settings["languages"]) 
	{
		result.Add(Language.Parse(lang.Name));
	}

  return result;
}
```

Or is there another way I didn't see to do this?

Thank you in advance!

**Comment by sergeyshushlyapin** [comment]
@arminsandorf, thanks for reporting this issue. The hardcoded language was sufficient for quite a long time but it seems it's time to extend it ;)

I wouldn't like to deal with settings and configs here since it's quite challenging to set up in unit tests. In some cases one may want to unit test behavior with 4 languages configured, in some other cases only one (default) language might be required. To achieve this the languages should be configured in the [Arrange](https://msdn.microsoft.com/en-us/library/hh694602.aspx#Anchor_3) section of a unit test:

```csharp
using (var db = new Db()
    .WithLanguages(Language.Parse("en"), Language.Parse("da")))
{
    var languages = db.Database.Languages;
    Assert.Contains(Language.Parse("en"), languages);
    Assert.Contains(Language.Parse("da"), languages);
}
```

// cc: @pveller

**Comment by sergeyshushlyapin** [comment]
@arminsandorf, @pveller any thoughts about the suggested solution? I'm about to complete the PR but would like to be sure it solves the problem.

There is might be another _more static_ way of configuration, e.g. one suggested by @arminsandorf or using some `Sitecore.FakeDb.DatabaseLanguages` setting containing all the allowed pipe-separated languages. As I stated above, it might not be handy in tests if there are numerous scenarios need to be covered, but might be useful if one needs to configure the _default_ set of languages once for all the tests.

Thanks.

**Comment by arminsandorf** [comment]
Hi Sergey,

thank you for this quick solution! For my needs the '.WithLanguages()' method works perfectly :) 

I thought about it and as you already mentioned, configure the languages using the App.config is not a good idea because it's too static. I think your solution provides more flexibility and also covers that '_default set of languages_'-case. So imho the PR can be completed.

**Comment by pveller** [comment]
I like the feel of `WithLanguages()` but at the same time I feel like those who need it will have to repeat it over and over again. Maybe do both? The nice fluid method on the `Db` for one-off tests and a configuration patch for a more _stable_ addition of languages to a whole suite of tests?

**Comment by sergeyshushlyapin** [comment]
That is exactly what I was thinking about last couple of days, thanks @pveller for highlighting this area! The _config patching_ is an excellent idea which BTW may be used later on to extend the default set of languages (without changes in FakeDb).

From the other side, I wouldn't like to introduce any new functionality until it's required. So, if @arminsandorf  is [satisfied](https://github.com/sergeyshushlyapin/Sitecore.FakeDb/issues/161#issuecomment-273055859) with the existing `WithLanguages()` method, I'd keep it as it is for the time being. Of course, I'm OK with extending it in future if necessary.

**Comment by sergeyshushlyapin** [comment]
Published as [v1.4.0](https://www.nuget.org/packages/Sitecore.FakeDb/).
